### ISSUE TITLE
`use_build_context_synchronously` wrongly warns

### ISSUE THREAD

**Issue by fzyzcjy** [dart-lang/linter#4472]
```dart
  Future<void> handleTap(BuildContext context) async {
    final result = await Navigator.pushNamed(context, '/hello') as int?;
  }
```

yields error while it should not:

![image](https://github.com/dart-lang/linter/assets/5236035/1b415f4d-10ef-4f97-9a35-62e26f0bbe37)

**Comment by srawlins** [comment]
Thanks for the report! Can you say what version of Dart/Flutter was running which reported this diagnostic? I suspect this is fixed at HEAD.

**Comment by fzyzcjy** [comment]
@srawlins https://github.com/fzyzcjy/flutter/commit/5182697c0dd2d0c7ad85c2dd3027374810f51f46

**Comment by srawlins** [comment]
Cool that's recent. Thanks for the report.

**Comment by fzyzcjy** [comment]
You are welcome!
